code/Weapons/Gadget/Gadget.cs

```diff
@@ -23,6 +23,8 @@ public partial class Gadget : AnimatedEntity, IResolvable
 
 	public bool Resolved => IsResolved();
 
+	public bool IsCrateGadget => Components.GetAll<GadgetComponent>().Any( c => c is CrateGadgetComponent );

```

Comment:
`public bool IsCrate => Components.Get<CrateGadgetComponent>() is not null;`
Does this work instead?

Reply:
Good catch 👍